### ISSUE TITLE
how to build a graph level classification model

### ISSUE THREAD

**Issue by Bitliuliu** [pyg-team/pytorch_geometric#1192]
hi, thank you for yours work to provide such a great tool for us.
Recently I tried to use GCN to build a three-layer classification network model. In the course of training, I found that the model does not converge. The training error is always very large, in my 2 classification task, the training accuracy rate is always around 50 percent, occasionally more than 70 percent.

my model  is constructed as below:
class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = GATConv(96, 60
                             )
        self.conv2 = GATConv(60, 30
                             )
        self.conv3 = GATConv(30, 10
                             )
        self.lin1 = torch.nn.Linear(10,2)
       
    def forward(self,data):
        x, edge_index,batchs = data.x, data.edge_index,data.batch
        x = F.relu(self.conv1(x, edge_index))
        x = F.dropout(x, training=self.training)
        x = F.relu(self.conv2(x, edge_index))
        x = F.dropout(x, training=self.training)
        x = F.relu(self.conv3(x, edge_index))
        x = F.dropout(x, training=self.training)
        x = global_mean_pool(x,batchs)
      
        x = self.lin1(x)
        return F.log_softmax(x, dim=1)

is there any clue for me to solve this problem?
thank you very much!!!

**Comment by Bitliuliu** [comment]
In practice, I tested both methods of convolution layer(GATconv, GCNconv), but neither seemed to work.

**Comment by rusty1s** [comment]
Well, that can have multiple reasons. Maybe you are even missing an `optimizer.step()` call? I would first remove dropouts and see if that fixes the issues. You can also replace the `GCNConv` with linear layers to see if that works.
